python/ray/util/spark/utils.py

```diff
@@ -383,18 +382,17 @@ def acquire_lock(file):
         mode = os.O_RDWR | os.O_CREAT | os.O_TRUNC
         file_ref = os.open(file, mode)
         locked_file_ref = None
-        lock_time = 10.0
-        start = current = time.time()
-        while current < start + lock_time:
+        # Allow for retrying getting a file lock a maximum number of seconds
+        max_lock_iter = 60 * 2
+        for _ in range(max_lock_iter):
             try:
-                fcntl.flock(file_ref, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                fcntl.flock(file_ref, fcntl.LOCK_EX)
             except (IOError, OSError):
                 pass
             else:
                 locked_file_ref = file_ref
                 break
             time.sleep(1.0)
-            current = time.time()
         if locked_file_ref is None:
             os.close(file_ref)
```

Comment:
If acquiring lock failed (timeout), let's delete the lock file, and create a new same name file, and acquiring lock on the new file.

This is for handling the case: sometimes user might cancel spark job that causes the python task process killed, in this case, the file might be locked forever (the process to releasing it has been killed) .
So the fallback code is for handling this case.

Reply:
@BenWilson2 This comment haven't been addressed ?